File: PUT.js
// A Program to send a PUT Request and set Block
const coap = require('coap');

//set the telemetry requerst for server
const req = coap.request({
    host:"coap.thingsboard.cloud",
    port: 5683,
    pathname: "/api/v1/aj6notRAvpht8n9Z8YTr/telemetry",
    // method: 'GET'
    method:'POST'
})
//setting Headrers for COAP
req.setOption("Content-Format","application/json")
req.setOption("Block2",Buffer.alloc(0x2))

//add payload
req.write("{'name':'Hello'}")

  req.on('response', (res) => {
    console.log(res.code)
    res.pipe(process.stdout)
    res.on('end', () => {
      process.exit(0)
    })
  })

  req.end()

File: PUT.py
import logging
import asyncio

from aiocoap import *

logging.basicConfig(level=logging.INFO)

async def main():
    """Perform a single PUT request to localhost on the default port, URI
    "/other/block". The request is sent 2 seconds after initialization.

    The payload is bigger than 1kB, and thus sent as several blocks."""

    context = await Context.create_client_context()

    await asyncio.sleep(2)

    HOST = "coap.thingsboard.cloud:"
    PORT = "5683"
    PATH = "/api/v1/aj6notRAvpht8n9Z8YTr/telemetry"
    payload = {"temp":24.654}

    request = Message(code=PUT, payload=payload, uri=HOST+PORT+PATH)

    response = await context.request(request).response

    print('Result: %s\n%r'%(response.code, response.payload))

if __name__ == "__main__":
    asyncio.run(main())
